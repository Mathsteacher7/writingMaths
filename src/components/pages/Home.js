import React from "react";

import Mathjax from "react-mathjax2";

const ascii = "U = 1/(R_si)";
const tex =
  "f(x) = \\int_{-\\infty}^\\infty\\hat f(\\xi)\\,e^{2 \\pi i \\xi x}\\,d\\xi";
const array = "\\begin{array}{cc} a & b \\\\ c & d \\end{array}";
const math = "\\frac{1}{\\sqrt{x^2 + 1}}";
const det = "x_{1,2}=\\frac{b\\pm\\sqrt{b^2-4ac}}{2}";
const dividing = "\\frac{1}{2}";
const equations = ["x+4=2", "3x=9", "2x-5=x+4"];

const Home = () => {
  return (
    <section className="section">
      <div className="container home" data-testid="homeTitle">
        hello world
      </div>
      <Mathjax.Context input="ascii">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{ascii}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{tex}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{array}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{math}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{math}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{math}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{det}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          This is an inline formula written in AsciiMath:{" "}
          <Mathjax.Node inline>{dividing}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          find x if we know that <Mathjax.Node>{dividing}</Mathjax.Node>
        </div>
      </Mathjax.Context>
      <br />
      <Mathjax.Context input="tex">
        <div>
          solve the equations:
          {equations.map((equation) => {
            return <Mathjax.Node>{equation}</Mathjax.Node>;
          })}
        </div>
      </Mathjax.Context>
    </section>
  );
};

export default Home;
