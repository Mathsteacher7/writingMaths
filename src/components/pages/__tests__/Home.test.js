import React from 'react';
import { render } from '@testing-library/react'
import Home from '../Home';

test('<Home />', () => {
    const { container, getByTestId } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('homeTitle').textContent).toBe('hello world')
})