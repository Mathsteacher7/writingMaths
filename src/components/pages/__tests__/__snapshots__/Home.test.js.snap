// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> 1`] = `
<section
  class="section"
>
  <div
    class="container home"
    data-testid="homeTitle"
  >
    hello world
  </div>
  <br />
  <br />
  <br />
</section>
`;
